From d91131ed7d58994db38d0c83e9b8249d4ecde289 Mon Sep 17 00:00:00 2001
From: Dominggoes Isakh <drjisakh@gmail.com>
Date: Thu, 27 Oct 2022 00:02:44 +0200
Subject: [PATCH 23/23] CameraService: Only warn if callingPid should be
 rejected to connect

Sometimes the CameraProvider dies but restart with new Pid. This
sometimes happens when switching between front/back camera. Just warn
but don't block connecting to the new CameraProvider.
---
 services/camera/libcameraservice/CameraService.cpp | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/services/camera/libcameraservice/CameraService.cpp b/services/camera/libcameraservice/CameraService.cpp
index ca2cfdc..39a15dc 100644
--- a/services/camera/libcameraservice/CameraService.cpp
+++ b/services/camera/libcameraservice/CameraService.cpp
@@ -1273,17 +1273,13 @@ Status CameraService::validateClientPermissionsLocked(const String8& cameraId,
     clientPid = callingPid;
 
     userid_t clientUserId = multiuser_get_user_id(clientUid);
-
     // Only allow clients who are being used by the current foreground device user, unless calling
     // from our own process OR the caller is using the cameraserver's HIDL interface.
     if (getCurrentServingCall() != BinderCallType::HWBINDER && callingPid != getpid() &&
             (mAllowedUsers.find(clientUserId) == mAllowedUsers.end())) {
-        ALOGE("CameraService::connect X (PID %d) rejected (cannot connect from "
+        ALOGW("CameraService::connect X (PID %d) should be rejected (should not connect from "
                 "device user %d, currently allowed device users: %s)", callingPid, clientUserId,
                 toString(mAllowedUsers).string());
-        return STATUS_ERROR_FMT(ERROR_PERMISSION_DENIED,
-                "Callers from device user %d are not currently allowed to connect to camera \"%s\"",
-                clientUserId, cameraId.string());
     }
 
     return Status::ok();
-- 
2.7.4

